controller/appcontroller.go

```diff
@@ -611,7 +611,11 @@ func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Applic
 
 	objs := make([]*unstructured.Unstructured, 0)
 	for k := range objsMap {
-		if ctrl.shouldBeDeleted(app, objsMap[k]) && objsMap[k].GetDeletionTimestamp() == nil {
+		if objsMap[k].GetDeletionTimestamp() != nil {
+			logCtx.Infof("%d objects remaining for deletion", len(objsMap))
+			return objs, nil
+		}
+		if ctrl.shouldBeDeleted(app, objsMap[k]) {
```

Comment:
hi @jannfis, 
if we have objects pending deletion we want to wait for them to complete before we proceed with the next wave of deletion. Since this function is registered as an update handler, it will be invoked when the deletes are completed, and we can issue `kubectl.DeleteResource` for the next wave then.

added a comment!